### PR TITLE
Zero or Null transfer tonnage deletion

### PR DESCRIPTION
**Pull request by NateHogan**

### CODE REVIEW

File: src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs [hunk 1]
```diff
@@ -289,7 +289,7 @@ public async Task<List<NoteTonnage>> GetTonnageByIds(List<Guid> ids)
             }
 
             var itemsToRemove =
-                note.NoteTransferTonnage.Where(ntt => tonnages.All(t => t.NoteTonnageId != ntt.NoteTonnageId));
+                note.NoteTransferTonnage.Where(ntt => tonnages.All(t => t.NoteTonnageId != ntt.NoteTonnageId) || (((ntt.Received == 0.00m && ntt.Reused == 0.00m) || (ntt.Received == null && ntt.Reused == null)) && status.Equals(NoteStatus.Submitted)));
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs:292]: @NateHogan Wouldnt bother checking the reused values. If the Recieved is null or zero just remove as reused can't have value if received doesnt

File: src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs [hunk 2]
```diff
@@ -322,5 +322,18 @@ public Task<bool> HasApprovedWasteHouseHoldEvidence(Guid recipientId, int compli
                                n.Status.Value == NoteStatus.Approved.Value &&
                                n.WasteType.Value == WasteType.HouseHold);
         }
+
+        public Note DeleteZeroTonnageFromSubmittedTransferNote(Note note, NoteStatus status, NoteType type)
+        {
+            if (status.Equals(NoteStatus.Submitted) && type.Equals(NoteType.TransferNote))
+            {
+                var itemsToRemove =
+                    note.NoteTransferTonnage.Where(ntt => (ntt.Received == 0.00m && ntt.Reused == 0.00m) || (ntt.Received == null && ntt.Reused == null));
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs:337]: @NateHogan  Wouldnt bother checking the reused values. If the Recieved is null or zero just remove as reused can't have value if received doesnt

File: src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs (hunk 1 above)
Comment (NateHogan) [reply src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs]: Updated

File: src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs (hunk 2 above)
Comment (NateHogan) [reply src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs]: Updated

File: src/EA.Weee.RequestHandlers/AatfEvidence/CreateTransferEvidenceNoteRequestHandler.cs [hunk 3]
```diff
@@ -65,6 +65,18 @@ public async Task<Guid> HandleAsync(TransferEvidenceNoteRequest request)
                         t.FirstTonnage,
                         t.SecondTonnage)).ToList();
 
+                    if (request.Status.Equals(Core.AatfEvidence.NoteStatus.Submitted))
+                    {
+                        for (int i = transferNoteTonnages.Count - 1; i >= 0; i--)
```
[the remark this answers is not in the document]
Comment (NateHogan) [reply src/EA.Weee.RequestHandlers/AatfEvidence/CreateTransferEvidenceNoteRequestHandler.cs]: Updated

File: src/EA.Weee.RequestHandlers/AatfEvidence/SetNoteStatusRequestHandler.cs [hunk 4]
```diff
@@ -35,6 +39,8 @@ public async Task<Guid> HandleAsync(SetNoteStatusRequest message)
                 ? evidenceNote.Organisation.Id
                 : evidenceNote.Recipient.Id);
 
+            evidenceDataAccess.DeleteZeroTonnageFromSubmittedTransferNote(evidenceNote, message.Status.ToDomainEnumeration<Domain.Evidence.NoteStatus>(), evidenceNote.NoteType);
```
[the remark this answers is not in the document]
Comment (NateHogan) [reply src/EA.Weee.RequestHandlers/AatfEvidence/SetNoteStatusRequestHandler.cs]: Updated

**Pull request merged**